### PR TITLE
Create silence and unsilence command

### PR DESCRIPTION
**Pull request by ks129**
Created commands for channel silencing and unsilencing. Supports both timed silencing and permanent silencing.

### CODE REVIEW

**Review by Akarys42 — CHANGES_REQUESTED** [review]
It looks pretty good so far. The main issue right now is that the bot will let you silence channels in which `@Developer` doesn't have any permission, or send is explicitly denied, the bot should instead send an error message.

File: src/main/kotlin/com/kotlindiscord/bot/extensions/SilenceExtension.kt [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        taskMapping.remove(taskData.channel.id.value)
+    }
+
+    override suspend fun setup() {
+        data class SilenceArguments(
+            val duration: List<String> = listOf()
+        )
+
+        command {
+            name = "silence"
```
[the remark this answers is not in the document]
Comment (gdude2002) [reply src/main/kotlin/com/kotlindiscord/bot/extensions/SilenceExtension.kt:48]: We do not.

File: src/main/kotlin/com/kotlindiscord/bot/extensions/SilenceExtension.kt [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                            denied += Permission.SendMessages
+                        }
+                        channel.createMessage(":lock: Successfully silenced channel for undefined time.")
+                    }
+                }
+            }
+        }
+
+        command {
+            name = "unsilence"
```
[the remark this answers is not in the document]
Comment (gdude2002) [reply src/main/kotlin/com/kotlindiscord/bot/extensions/SilenceExtension.kt:109]: Nop, no hide pls.

**Comment by gdude2002** [comment]
While I appreciate the work done on this, and it's not your fault that bot development stalled (ultimately I just had far too much to do), I now have a better understanding of how to write this type of extension - and the bot needs to be updated to the latest version of the framework before any more development can continue, so it would be best to close this now and rewrite it later on.

Even so, thanks for spending time on this!

**Pull request closed without merging**